ref: Restructure options

limbonaut opened the pull request:
This PR restructures options in the Project Settings:
- Mark options as basic and advanced to have a cleaner interface.
- Move error logger tunables into their own sub-page. This is a **BREAKING** change so make sure to reapply those error logger values if you're changing the defaults.
  - Rename options: `error_logger_*` => `logger_*`.

#### Basic options

<img width="718" alt="Screenshot 2025-04-04 at 17 40 51" src="https://github.com/user-attachments/assets/3e794274-2f2a-4f9f-ad21-d85e44c86910" />
<img width="716" alt="Screenshot 2025-04-04 at 17 41 02" src="https://github.com/user-attachments/assets/6a3fd2ee-a38a-4f9b-8211-b161cb924169" />

#### Advanced options

<img width="721" alt="Screenshot 2025-04-04 at 17 41 27" src="https://github.com/user-attachments/assets/2d286004-b793-497b-8782-fb28c42edd36" />

<img width="719" alt="Screenshot 2025-04-04 at 17 41 42" src="https://github.com/user-attachments/assets/5c7d6474-782f-4e68-8aab-da22c780e5a1" />

#### Options API

<img width="518" alt="Screenshot 2025-04-04 at 17 48 02" src="https://github.com/user-attachments/assets/05c7fc45-85fe-4498-97ea-67fa632a4bef" />

bitsandfoxes reviewed (commented):
This looks so much cleaner now! 🔥 

I'm assuming a user here: Just from the option's name `"Error Logger" -> "Enabled"` it's not clear to me what that is or what it does. Does Sentry use one? Is it replacing Godot's? 
What about naming it `(Automatic?) Log Capture`, or `Logging Integration`?
Also, can we add a text-field to the options? Could be just a one-liner: "The SDK reads the log file (or hooks up into the logger in the future), parses log messages and sends those errors to Sentry".

limbonaut commented:
@bitsandfoxes Name is constructed from the setting keys we supply, but we can provide any key we want there, for example: 
- `sentry/error_logger/error_logger_enabled` key will transform to `"Error Logger" page -> "Error Logger Enabled" field`

We don't need to have the same names as in `SentryOptions` class, so there is some flexibility, but it's a good idea to keep it synchronized. We group all error logger related options under the `error_logger_` prefix in the options. This way, it's easy to communicate relation between them, or put them under the same category/group (we need to have a common prefix for this). I'm open to shortening it to `logger_*` if we want shorter names. `Log Capture` would clash with `Attach Log File` which would be confusing. And since we're expecting a `Logger` class [to be exposed in Godot](https://github.com/godotengine/godot/pull/91006/files#diff-0b36b3b54d28ebf01241affa02c98cd792676198da8a703c4b3cdd1ce5e784b4), I recommend sticking with the `Logger` prefixed names here. As we're keeping it compatible with the expected Godot changes, it will be a straightforward replacement when this functionality lands in Godot, reusing the same option keys.

As for some kind of a hint field, or text field, there is no such built-in option. I think we could develop something like that, but it's certainly wouldn't be trivial, as it would involve making an Inspector Editor plugin and property editor classes: maybe 100+ lines just to be able to do it in places. It may be possible, and I can look into it if you think it's worth exploring this idea.

In conclusion, I see the point of just `Enabled` being confusing, as it needs a little bit more context - I'll address that, keeping to the common prefix. And maybe shorten the error logger option keys to simply starting with `logger_`?

limbonaut commented:
I've updated the hat with the new screenshots. `logger_enabled` option has improved text and `error_logger_*` options shortened to `logger_*` . I also added a screenshot from the class docs showing how the API looks like with these changes.
